core/os/os2/process_linux.odin

```diff
@@ -1,68 +1,299 @@
+
+			n = strings.index_byte(passwd, ':')
+			uid: int
+			ok: bool
+			if uid, ok = strconv.parse_int(passwd[:n]); ok && uid == int(s.uid) {
+				info.username = strings.clone(username, allocator) or_return
+				break
+			} else if !ok {
+				return info, .Invalid_File
+			}
+
+			eol := strings.index_byte(passwd, '\n')
+			if eol == -1 {
+				break
+			}
+			passwd = passwd[eol + 1:]
+		}
+	}
+
+	cmdline_if: if selection & {.Working_Dir, .Command_Line, .Command_Args, .Executable_Path} != {} {
+		path_len = len(fmt.bprintf(path_slice, "/proc/%d/cmdline", pid))
+		path_backing[path_len] = 0
+
+		cmdline_bytes := _read_entire_pseudo_file(path_cstr, temp_allocator()) or_return
+		if len(cmdline_bytes) == 0 {
+			break cmdline_if
+		}
+		cmdline := string(cmdline_bytes)
+
+		terminator := strings.index_byte(cmdline, 0)
+
+		command_line_exec := cmdline[:terminator]
+
+		// Still need cwd if the execution on the command line is relative.
+		cwd: string
+		cwd_err: Error
+		if .Working_Dir in selection || (.Executable_Path in selection && command_line_exec[0] != '/') {
+			path_len = len(fmt.bprintf(path_slice, "/proc/%d/cwd", pid))
+			path_backing[path_len] = 0
+
+			cwd, cwd_err = _read_link_cstr(path_cstr, temp_allocator()) // allowed to fail
+			if cwd_err == nil && .Working_Dir in selection {
+				info.working_dir = strings.clone(cwd, allocator) or_return
+			}
+		}
+
+		if .Executable_Path in selection {
+			if cmdline[0] == '/' {
+				info.executable_path = strings.clone(cmdline[:terminator], allocator) or_return
+			} else if cwd_err == nil {
+				join_paths: [2]string = { cwd, cmdline[:terminator] }
+				info.executable_path = filepath.join(join_paths[:], allocator)
+			}
+		}
+		if .Command_Line in selection {
+			info.command_line = strings.clone(cmdline[:terminator], allocator) or_return
+		}
+
+		if .Command_Args in selection {
+			// skip to first arg
+			cmdline = cmdline[terminator + 1:]
+
+			arg_list := make([dynamic]string, allocator) or_return
+			for len(cmdline) > 0 {
+				terminator = strings.index_byte(cmdline, 0)
+				arg := strings.clone(cmdline[:terminator], allocator) or_return
+				append(&arg_list, arg) or_return
+				cmdline = cmdline[terminator + 1:]
+			}
+			info.command_args = arg_list[:]
+		}
+	}
+
+	stat_if: if selection & {.PPid, .Priority} != {} {
+		path_len = len(fmt.bprintf(path_slice, "/proc/%d/stat", pid))
+		path_backing[path_len] = 0
+
+		proc_stat_bytes := _read_entire_pseudo_file(path_cstr, temp_allocator()) or_return
+		if len(proc_stat_bytes) <= 0 {
+			break stat_if
+		}
+
+		start := strings.last_index_byte(string(proc_stat_bytes), ')')
+		stats := string(proc_stat_bytes[start + 2:])
+
+		// We are now on the 3rd field (skip)
+		stats = stats[strings.index_byte(stats, ' ') + 1:]
+
+		if .PPid in selection {
+			ppid_str := stats[:strings.index_byte(stats, ' ')]
+			if ppid, ok := strconv.parse_int(ppid_str); ok {
+				info.ppid = ppid
+			} else {
+				return info, .Invalid_File
```

Comment:
I assume this only happens when it is an invalid thing? If it isn't then you might want to set the error and break to allow retrieval of other fields.